File: python/softeer/hanyang_hcpc_2023.py
# Hanyang Popularity Exceeding Competition

import sys

input = sys.stdin.readline

if __name__ == "__main__":
    n = int(input())
    x = 0
    for _ in range(n):
        p,c = map(int, input().split())
        if abs(p - x) <= c:
            x += 1
    print(x)

File: python/softeer/Lv3/hanyang_hcpc_2023.cpp
// Hanyang Popularity Exceeding Competition

#include<iostream>

using namespace std;

int main(int argc, char** argv)
{
    int n;
    std::cin >> n;
    int x = 0;
    int p = 0, c = 0;
    for(int i = 0; i < n; i++){
        std::cin >> p >> c;
        if(abs(p - x) <= c){
            x++;
        }
    }
    std::cout << x;
    return 0;
}